apps/aecore/src/aec_peers.erl

```diff
@@ -672,27 +672,34 @@ ping_peer(Peer) ->
     %% needed for the ping itself, but given that a ping can quickly
     %% lead to a greater discovery, we should be prepared to handle pings
     %% ourselves at this point.
-    await_aehttp(),
-    Res = aeu_requests:ping(Peer),
-    lager:debug("ping result (~p): ~p", [Peer, Res]),
-    case Res of
-        {ok, Map} ->
-            log_good_ping(Peer#peer.uri),
-            Peers = maps:get(<<"peers">>, Map, []),
-            add_and_ping_peers(Peers),
-            %% [add(P, true) || P <- Peers],
-            ok;
-        _ ->
-            log_ping_error(Peer#peer.uri),
-            ok
+    case await_aehttp() of
+        ok ->
+            Res = aeu_requests:ping(Peer),
+            lager:debug("ping result (~p): ~p", [Peer, Res]),
+            case Res of
+                {ok, Map} ->
+                    log_good_ping(Peer#peer.uri),
+                    Peers = maps:get(<<"peers">>, Map, []),
+                    add_and_ping_peers(Peers);
+                _ ->
+                    log_ping_error(Peer#peer.uri)
+            end;
+        {error, timeout} ->
+            lager:debug("timeout waiting for aehttp - no ping (~p) will retry", [Peer#peer.uri]),
+            ping_peer(Peer)
```

Comment:
no retry counter?

Reply:
Not sure what would be the right action when we reach that counter, better to keep trying?